packages/quantic/force-app/main/default/lwc/quanticDidYouMean/e2e/quanticDidYouMean.e2e.ts

```diff
@@ -0,0 +1,237 @@
+import {useCaseTestCases} from '../../../../../../playwright/utils/useCase';
+import {testInsight, testSearch, expect} from './fixture';
+import mockData from './data';
+
+const {
+  originalQuery: expectedOriginalQuery,
+  correctedQuery: expectedCorrectedQuery,
+  didYouMeanData,
+  didYouMeanNextData,
+  queryTriggerData,
+  triggeredQuery: expectedTriggeredQuery,
+} = mockData;
+
+const fixtures = {
+  search: testSearch,
+  insight: testInsight,
+};
+
+useCaseTestCases.forEach((useCase) => {
+  let test = fixtures[useCase.value];
+
+  test.describe(`quantic did you mean ${useCase.label}`, () => {
+    test.describe('when we have an automatic did you mean query correction', () => {
+      test.describe('with queryCorrectionMode set to "legacy"', () => {
+        test.use({
+          didYouMeanData,
+          options: {
+            disableQueryAutoCorrection: false,
+            queryCorrectionMode: 'legacy',
+          },
+        });
+        test('should display the didYouMean component and send the analytics', async ({
+          didYouMean,
+          search,
+        }) => {
+          await search.mockSearchWithDidYouMeanResponse(didYouMeanData);
+          const searchResponsePromise = search.waitForSearchResponse();
+          const automaticallyCorrectSearchPromise =
+            search.waitForSearchResponse();
+          const expectedUaRequestPromise =
+            didYouMean.waitForDidYouMeanAutomaticAnalytics({
+              queryText: expectedCorrectedQuery,
+            });
+          didYouMean.setQuery(expectedOriginalQuery);
+          didYouMean.performSearch();
+          await searchResponsePromise;
+          await expectedUaRequestPromise;
+          await automaticallyCorrectSearchPromise;
+
+          await expect(didYouMean.didYouMeanNoResultsLabel).toBeVisible();
+          await expect(didYouMean.didYouMeanNoResultsLabel).toContainText(
+            expectedOriginalQuery
+          );
+          await expect(
+            didYouMean.didYouMeanAutomaticQueryCorrectionLabel
+          ).toBeVisible();
+          await expect(
+            didYouMean.didYouMeanAutomaticQueryCorrectionLabel
+          ).toContainText(expectedCorrectedQuery);
+        });
+      });
+
+      test.describe('with queryCorrectionMode set to "next"', () => {
+        test.use({
+          didYouMeanData,
+          options: {
+            disableQueryAutoCorrection: false,
+            queryCorrectionMode: 'next',
+          },
+        });
+        test('should display the didYouMean component and send the analytics', async ({
```

Comment:
same here, why not checking the rendering in the unit tests?

Reply:
Same answer as above :)